core/src/main/java/io/questdb/griffin/engine/functions/groupby/FirstStrGroupByFunction.java

```diff
@@ -52,7 +52,7 @@ public void clear() {
     @Override
     public CharSequence getStr(Record rec) {
         boolean hasStr = rec.getBool(valueIndex);
-        return hasStr ? sink : null;
+        return hasStr ? sink.toString() : null;
```

Comment:
We try to avoid allocations as much as possible. That's why `getStr` and `getSym` methods return a flyweight `CharSequence` instead of a `String`.
```suggestion
        return hasStr ? sink : null;
```

Reply:
updated